validator_derive/src/asserts.rs

```diff
@@ -113,7 +113,7 @@ pub fn assert_has_len(field_name: String, type_name: &str, field_type: &syn::Typ
         return;
     }
 
-    if type_name != "String"
+    if !type_name.ends_with("String")
         && !type_name.starts_with("Vec<")
         && !type_name.starts_with("Option<Vec<")
         && !type_name.starts_with("Option<Option<Vec<")
```

Comment:
should we just search for String and str rather than this ever-expanding list of possibilities? Worst case, it will be a compile time error if they tried to use it on a type it can't be used but that can be documented in the README.

Reply:
it does sound like a sensible approach to me. The validation error feels like a very nice message for easier development, but if it's getting too much on the way then relaxing it would help too.

I've made the change to simplify this as much as possible. There are still `Option<Option<` checks for other types but they don't affect the case of strings

Reply:
We should keep the same logic for the length validator below though